unix/vncserver/vncsession.c

```diff
@@ -511,14 +511,23 @@ main(int argc, char **argv)
 
     const char *username, *display;
 
-    if ((argc != 3) || (argv[2][0] != ':')) {
+    int forking = 1;
+    if (argc == 4) {
+        if ((getopt(argc, argv, "D") == 'D') && (argv[3][0] == ':'))
+            forking = 0;
+        else {
+            fprintf(stderr, "Syntax:\n");
+            fprintf(stderr, "    %s [-D] <username> <display>\n", argv[0]);
+            return EX_USAGE;
```

Comment:
Let's move this to a helper function so we can avoid duplication.

Reply:
Also, no problem.

Reply:
> Let's move this to a helper function so we can avoid duplication.

I did not move the 'return EX_USAGE' statement to the helper function, usage(), because I did not understand the implications of returning the EX_USAGE value from usage() instead of returning it from main. Is that OK?

Reply:
> > Let's move this to a helper function so we can avoid duplication.
> 
> I did not move the 'return EX_USAGE' statement to the helper function, usage(), because I did not understand the implications of returning the EX_USAGE value from usage() instead of returning it from main. Is that OK?

OK, I figured out how to do it, I used _exit(EX_USAGE) in the new helper function usage().